src/tests/conformance/post_handshake/msg_of_interest.rs

```diff
@@ -0,0 +1,45 @@
+use tempfile::TempDir;
+
+use crate::{
+    protocol::codecs::tagmsg::Payload, setup::node::Node,
+    tools::synthetic_node::SyntheticNodeBuilder,
+};
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn c005_MSG_OF_INTEREST_expect_after_connect() {
+    // ZG-CONFORMANCE-005
+
+    crate::tools::synthetic_node::enable_tracing();
+    // Spin up a node instance.
+    let target = TempDir::new().expect("couldn't create a temporary directory");
+    let mut node = Node::builder()
+        .log_to_stdout(true)
+        .build(target.path())
+        .expect("unable to build the node");
+    node.start().await;
+
+    // Create a synthetic node and enable handshaking.
+    let mut synthetic_node = SyntheticNodeBuilder::default()
+        .build()
+        .await
+        .expect("unable to build a synthetic node");
+
+    let net_addr = node.net_addr().expect("network address not found");
+
+    // Connect to the node and initiate the handshake.
+    synthetic_node
+        .connect(net_addr)
+        .await
+        .expect("unable to connect");
+
+    let check = |m: &Payload| matches!(&m, Payload::MsgOfInterest(..));
+    assert!(synthetic_node.expect_message(&check).await);
+
+    // Gracefully shut down the nodes.
+    synthetic_node.shut_down().await;
+    node.stop().expect("unable to stop the node");
+}
+
+// TODO(Rqnsom): c005_t2: when the node initiates the connection, send MSG_OF_INTEREST with all messages enabled
```

Comment:
I could add this test as a separate commit. Not sure if we can split all the codecs into multiple commits.